### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by dimqua** [MrAlex94/Waterfox#234]
Please support an inclusion of the Waterfox (55.2 and later) for Android to the [F-Droid](https://f-droid.org/) repository. It would be beneficial for both the Waterfox's popularity and for those users who prefer don't use Google Play store.

I already posted a request here: https://gitlab.com/fdroid/rfp/issues/373

**Comment by grahamperrin** [comment]
Discussion: [When will Waterfox be available on F-Droid? : waterfox](https://www.reddit.com/r/waterfox/comments/8k1n3z/when_will_waterfox_be_available_on_fdroid/)

**Comment by benryanau** [comment]
Valued addition

**Comment by IzzySoft** [comment]
As for complexity: There are several variants of Firefox listed at F-Droid (e.g. Fennec). On the other hand, we are totally understaffed, so it might take a little to get it listed (depending on how hard it really is to build). But then, if the Waterfox team is interested, why not try and open an RFP (Request For Packaging)?

**Comment by tcreek** [comment]
How is that F-Droid inclusion coming along?

**Comment by IzzySoft** [comment]
@tcreek check at [the RFP](https://gitlab.com/fdroid/rfp/issues/373): F-Droid currently waits for removal of proprietary components, here: GCM, being confirmed by @MrAlex94, as those are not permitted by the inclusion policy.

**Comment by tcreek** [comment]
Then it would probably be easier if there was just an easy to download listed link on the main site along with the other downloads listed.

**Comment by IzzySoft** [comment]
Easier may be, but not preferable (though OK as a work around until it's fixed). Having it in the F-Droid catalog makes it easier to find, and also adds a level of trust.

**Comment by tcreek** [comment]
> @tcreek check at [the RFP](https://gitlab.com/fdroid/rfp/issues/373): F-Droid currently waits for removal of proprietary components, here: GCM, being confirmed by @MrAlex94, as those are not permitted by the inclusion policy.

Are you sure about that? I was just looking around on F-Droid and see one application which has a caveat, "This app promotes non-free network services." It is called "Weather Widget."

**Comment by IzzySoft** [comment]
@tcreek being one of the F-Droid maintainers: yes, I'm sure about that :wink: Don't confuse "NonFreeNet" (uses/promotes non-free network services) with "contains proprietary elements" (no AntiFeature for that as it is not acceptable by the inclusion policy at all; F-Droid always builds from source only, so no JARs etc. allowed – and GCM/FCM requires Google Services which are proprietary/non free).

Weather Widget probably uses weather data from WU, Yahoo, OW or some other non-free service – but the app itself can be built completely from source.

**Comment by Serkan-devel** [comment]
I've seen some Browsers without

**Comment by Serkan-devel** [comment]
I've seen some browsers on fdroid without any antifeatures listed

**Comment by IzzySoft** [comment]
@Serkan-devel of course, why not? As long as they do not access nonfree network resources *automatically* or even depend on them, no reason to flag them. There's a difference between an app like Yalp (which solely exists to access a nonfree network service, namely Play Store) and a "clean" web browser (where the user isn't forced to access any nonfree service – well, at least not by the browser). So what AntiFeature would you expect every web browser to have? Just wondering where there might be another misunderstanding I might help solving :smiley:

**Comment by tcreek** [comment]
After the statement from F-Droid, I would never put anything on that site. I say close this request.

https://f-droid.org/en/2019/07/16/statement.html

**Comment by TPS** [comment]
@tcreek To be clear, to what precisely do you object in that statement that would be relevant to _this_?

**Comment by tcreek** [comment]
The fact they have taken a political stance("-Droid is taking a political stance here." against so-called "oppression or harassment against marginalized groups." & "oppression or harassment to happen at its communication channels. "

We see where that is happening now then they are the ones deciding what all that means. Like on various Alphabet platforms, Twitter, and Facebook.  When they are now becoming the arbiter of all that, rather than a neutral party, they are no longer any better than those tech companies many are wanting to get away from, and their biased censorship.

No thanks to that!

**Comment by TPS** [comment]
@tcreek You're clearly trolling, so I refuse to respond to you further.

For those who might be swayed by the above:  So, F-Droid _also_ strictly requires that apps in their repo be FOSS software built & signed w/ their own key, which is no less a political stance than not allowing people to be abused on their media platforms.  For many of us, neither is an issue.  That _still_ doesn't prevent us from wanting Waterfox on F-Droid.

IMHO, the practical ways to get this done are:
- Do all the work to comply w/ F-Droid's guidelines
- Roll out a Waterfox-controlled F-Droid-compatible repo for distributing APKs
- Get included in another repo (e.g., @rfc2822's [Unofficial F-Droid repo for various apps](https://rfc2822.gitlab.io/fdroid-firefox/fdroid/repo?fingerprint=8F992BBBA0340EFE6299C7A410B36D9C8889114CA6C58013C3587CDA411B4AED), which started as a way to distribute various official browser APKs

**Comment by Mikaela** [comment]
> So, F-Droid also strictly requires that apps in their repo be FOSS software built & signed w/ their own key

If I understand correctly, it's possible to also use the upstream key as long as the builds are reproducible.

* https://f-droid.org/en/docs/Reproducible_Builds/

**Comment by grahamperrin** [comment]
> The fact they have taken a political stance …

As people may object to a stance there, so people may object to a stance here. 

Let's keep this issue focused on Waterfox, not politics. Thanks.

**Comment by tcreek** [comment]
It's not talking politics, it is keeping away from politics by not supporting them for their political stance, and potential to abuse with censorship.

**Comment by grahamperrin** [comment]
> … potential to abuse with censorship. …

https://www.thesaurus.com/browse/censorship?s=t … _suppression_ … 

@tcreek whilst disliking potential censorship you encourage suppression of a method of distribution. 

Emphatically: let's keep this issue focused on **Waterfox**. Not politics, not censorship, not word games …

**Comment by tcreek** [comment]
They already are , per their statement, engaging in suppression of distribution.  It is pertinent here because this very thread discusses distributing it on that platform.

**Comment by grahamperrin** [comment]
@MrAlex94 maybe time to begin using the _off-topic_ feature of GitHub? Thanks.

**Comment by Korsola** [comment]
Any updates on this?

**Comment by yashpalgoyal1304** [comment]
https://github.com/WaterfoxCo/Waterfox/issues/143

**Comment by VespaFerox** [comment]
Now that [Waterfox](https://www.waterfox.net) is available via _Google Play_, the **F-Droid** version **shouldn't be long in coming**, _right_?
